runtime/cpp/yolov5seg/yolov5seg_example.cpp

```diff
@@ -0,0 +1,405 @@
+
+    std::chrono::duration<double> total_time;
+    std::chrono::time_point<std::chrono::system_clock> t_start = std::chrono::high_resolution_clock::now();
+
+    std::string yolo_hef       = getCmdOption(argc, argv, "-hef=");
+    std::string video_path      = getCmdOption(argc, argv, "-input=");
+
+    std::chrono::time_point<std::chrono::system_clock> write_time_vec;
+    std::chrono::duration<double> inference_time;
+    std::chrono::duration<double> postprocess_time;
+
+    auto vdevice_exp = VDevice::create();
+    if (!vdevice_exp) {
+        std::cerr << "Failed create vdevice, status = " << vdevice_exp.status() << std::endl;
+        return vdevice_exp.status();
+    }
+    auto vdevice = vdevice_exp.release();
+
+    auto network_group = configure_network_group(*vdevice, yolo_hef);
+    if (!network_group) {
+        std::cerr << "Failed to configure network group " << yolo_hef << std::endl;
+        return network_group.status();
+    }
+
+    auto input_vstreams_params = network_group.value()->make_input_vstream_params(QUANTIZED, FORMAT_TYPE_INPUT, HAILO_DEFAULT_VSTREAM_TIMEOUT_MS*100, HAILO_DEFAULT_VSTREAM_QUEUE_SIZE);
+    if (!input_vstreams_params) {
+        std::cerr << "Failed creating input vstreams " << input_vstreams_params.status() << std::endl;
+        return input_vstreams_params.status();
+    }
+
+    auto output_vstreams_params = network_group.value()->make_output_vstream_params(QUANTIZED, FORMAT_TYPE_OUTPUT, HAILO_DEFAULT_VSTREAM_TIMEOUT_MS*100, HAILO_DEFAULT_VSTREAM_QUEUE_SIZE);
+    if (!output_vstreams_params) {
+        std::cerr << "Failed creating vstreams " << output_vstreams_params.status() << std::endl;
+        return output_vstreams_params.status();
+    }
+
+    auto input_vstreams = VStreamsBuilder::create_input_vstreams(*network_group.value(), *input_vstreams_params);
+    if (!input_vstreams) {
+        std::cerr << "Failed creating input vstreams " << input_vstreams.status() << std::endl;
+        return input_vstreams.status();
+    }
+
+    auto output_vstreams = VStreamsBuilder::create_output_vstreams(*network_group.value(), *output_vstreams_params);
+    if (!output_vstreams) {
+        std::cerr << "Failed creating output vstreams " << output_vstreams.status() << std::endl;
+        return output_vstreams.status();
+    }
+
+    auto vstreams = std::make_pair(input_vstreams.release(), output_vstreams.release());
+
+    std::vector<std::chrono::time_point<std::chrono::system_clock>> read_time_vec(vstreams.second.size());
+
+    print_net_banner(vstreams);
+
+    cv::VideoCapture capture(video_path);
+    if (!capture.isOpened()){
+        throw "Error when reading video";
+    }
+    double frame_count = capture.get(cv::CAP_PROP_FRAME_COUNT);
+    double org_height = capture.get(cv::CAP_PROP_FRAME_HEIGHT);
+    double org_width = capture.get(cv::CAP_PROP_FRAME_WIDTH);
+    capture.release();
+
+    status = run_inference<uint16_t>(std::ref(vstreams.first), 
```

Comment:
instead of uint_16, use the datatype from hef (in this specific hef it will be indeed uint16, but make it general)